sockeye/lr_scheduler.py

```diff
@@ -128,7 +154,8 @@ def __call__(self, t):
         if self.lr is None:
             assert self.base_lr is not None
             self.lr = self.base_lr
-        return self.lr
+        lr = self._warmup(t) if self.warmup > 0 and t <= self.warmup else self.lr
```

Comment:
if the learning rate is reduced at any step `t < warmup` because the validation score plateaued you get the log
```
%d checkpoints since improvement or rate scaling, lowering learning rate: %1.2e -> %1.2e
```
which technically would not be true as the learning rate would still be coming from self._warmup rather then being self.lr, which could be slightly confusing. That's all I mean.